woocommerce/wc-admin
# Show task and activity notifications in the Inbox panel

Fixes #6737

This PR adds abbreviated cards to show task and activity notifications in the Inbox panel.

The original design changed a bit, so take the image in [this comment](https://github.com/woocommerce/woocommerce-admin/issues/6737#issuecomment-840831948) as a reference. 

Also, the Inbox panel width will be 80px smaller.

### Accessibility




### Screenshots

![screenshot-one wordpress test-2021 05 21-09_50_58](https://user-images.githubusercontent.com/1314156/119140942-1ab6c100-ba1b-11eb-889a-c284378bf0fa.png)

### Detailed test instructions:

-   Go to the `Orders` page (the important thing here is to go to any page other than `Home`).
-   Open the `Inbox` panel.
-   Verify the `Orders`, `Reviews`, and `Stock` abbreviated cards are not visible when the Setup tasks are visible in the `Home`screen.
-   Verify that the card `Things to do next` is only visible when the extended tasks are visible.
-   ~The `Store Alerts` are indicated as critical tasks (`# critical alerts`). To add a store alert (a critical task to show), you can use the plugin [wc-admin-test-helper](https://github.com/woocommerce/woocommerce-admin-test-helper) or change the type to `update` of an (unactioned and not deleted) admin note. You can use a SQL statement like this:~

```
UPDATE `wp_wc_admin_notes` SET `type` = 'update', `is_deleted` = 0, `status` = 'unactioned' WHERE `wp_wc_admin_notes`.`name` = 'wc-admin-onboarding-payments-reminder'
```

To return the note back to normal, you should execute a statement like:

```
UPDATE `wp_wc_admin_notes` SET `type` = 'info', `is_deleted` = 0, `status` = 'unactioned' WHERE `wp_wc_admin_notes`.`name` = 'wc-admin-onboarding-payments-reminder'
```

-   Click on every abbreviated card and verify the user is redirected to the `Home` screen and the selected panel is opened. In the case of `Things to do next`, the user should be redirected to the Home screen with the extended task list anchored to the top of the view.
-   Verify that the "unread indicator" in the Inbox Icon is active when there is an unread note or an abbreviated panel to show.
-   Enable debug messages inside browser devtools and verify the events are being recorded correctly. The event `wcadmin_activity_panel_click` should be recorded after clicking with the selected task as a prop.


### UPDATE ###
After chatting with @elizaan36 , the `critical alerts` tag is removed for now (commit cf65b63).
